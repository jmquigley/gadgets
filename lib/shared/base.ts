import * as React from 'react';
import {Size} from './index';

const styles = require('./styles.css');

export interface BaseOptions {
	className?: boolean;
	disabled?: boolean;
	visible?: boolean;
}

const defaultBaseOptions: BaseOptions = {
	className: true,
	disabled: true,
	visible: true
};

/**
 * A base class between React and all components in the module.
 */
export abstract class BaseComponent<P, S> extends React.Component<P, S> {

	protected _classes: string[] = [];
	protected _inlineStyle: any = {};    // inline style overrides
	protected _styles: any = {};         // css modules styles

	constructor(props: P, pstyles: any = {}) {
		super(props);
		this._styles = pstyles;
	}

	get classes(): string[] {
		return this._classes;
	}

	set classes(str: string[]) {
		this._classes = str;
	}

	get inlineStyle(): any {
		return this._inlineStyle;
	}

	set inlineStyle(val: any) {
		this._inlineStyle = Object.assign(this._inlineStyle, val);
	}

	get style(): string {
		return this._inlineStyle;
	}

	get styles(): any {
		return this._styles;
	}

	protected buildStyles(props: P, style: any = {}, opts?: BaseOptions): void {
		this._classes = [];
		this._inlineStyle = Object.assign(this._inlineStyle, props['style'], style);

		opts = Object.assign(
			defaultBaseOptions,
			opts
		);

		if (props['className'] !== '' && opts.className) {
			this._classes.push(props['className']);
		}

		if (!props['visible'] && opts.visible) {
			this._classes.push(this.styles.invisible);
		}

		if (props['disabled'] && opts.disabled) {
			this._classes.push(styles.disabled);
			this._classes.push('nohover');
		}
	}

	/**
	 * Takes a sizing parameter and a styles object and searches for the
	 * corresponding style class CSS.  If it is found, then it returns the local
	 * style for that size.
	 * @returns {string} the name of the local style for that size.  If
	 * it is not found, then an empty string is returned.
	 */
	protected getSizeStyle() {

		switch (this.props['size']) {
		case Size.xxsmall:
			if ('xxsmall' in this.styles) {
				return this.styles.xxsmall;
			}
			break;

		case Size.xsmall:
			if ('xsmall' in this.styles) {
				return this.styles.xsmall;
			}
			break;

		case Size.small:
			if ('small' in this.styles) {
				return this.styles.small;
			}
			break;

		case Size.large:
			if ('large' in this.styles) {
				return this.styles.large;
			}
			break;

		case Size.xlarge:
			if ('xlarge' in this.styles) {
				return this.styles.xlarge;
			}
			break;

		case Size.xxlarge:
			if ('xxlarge' in this.styles) {
				return this.styles.xxlarge;
			}
			break;

		case Size.normal:
		case Size.medium:
		default:
			if ('medium' in this.styles) {
				return this.styles.medium;
			}
		}

		return '';
	}
}
