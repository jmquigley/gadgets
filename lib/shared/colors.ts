export type ColorType = string;

export enum ColorScheme {
	c1 = "#004358",
	c2 = "#1f8a70",
	c3 = "#bedb39",
	c4 = "#ffe11a",
	c5 = "#fd7400"
}

export enum Color {
	black = "#000000",
	error = "#d9534f",
	glow = "#51cbee",
	gray = "#666666",
	info = "#5bc0de",
	lightgray = "#cccccc",
	lightsaber = "#2ff923",
	red = "#ff0000",
	silver = "#c0c0c0",
	slategray = "#708090",
	success = "#5cb85c",
	ultralightgray = "#e6e6e6",
	warning = "#f0ad4e",
	white = "#ffffff"
}
