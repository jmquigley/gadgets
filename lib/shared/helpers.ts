"use strict";

const debug = require("debug")("helpers");

/**
 * Takes a variable name and an object and places that name and associated
 * object into the global node space.  If the object is already in the
 * global space, then that reference is used and returned.  The replace
 * flag is used to override the current global reference.
 * @param name {string} the name of the variable to store globally
 * @param pkg {object} the object associated with the variable name
 * @param replace {boolean} if true, then it will always overwrite the
 * current global (if it exists)
 * @return {object} the global instance reference
 */
export function globalize(
	name: string,
	pkgname: any,
	replace: boolean = false
) {
	let ref: any = pkgname;
	if (!(global as any)[name] || replace) {
		(window as any)[name] = (global as any)[name] = pkgname;
		debug("Adding global variable %o -> %O", name, (global as any)[name]);
	} else {
		ref = (global as any)[name];
	}

	return ref;
}
