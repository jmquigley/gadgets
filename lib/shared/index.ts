import {
	BaseComponent,
	baseZIndex,
	defaultSize
} from './base';

import {
	getDefaultWrapperProps,
	Wrapper,
	WrapperProps
} from './wrapper';

import {
	BaseProps,
	Color,
	ColorScheme,
	Direction,
	disabled,
	DisabledCSS,
	getDefaultBaseProps,
	invisible,
	InvisibleCSS,
	Justify,
	Location,
	locationStyle,
	SortOrder,
	Styles
} from './props';

import {
	borderStyle,
	boxStyle,
	fontStyle,
	FontStyle,
	lineHeightStyle,
	rectStyle,
	Sizes,
	Sizing,
	Styling
} from './sizing';

import {
	BaseState,
	getDefaultBaseState
} from './state';

import {
	getTheme,
	getThemeList,
	setTheme,
	Theme,
	ThemeProps
} from './themes';

export {
	BaseComponent,
	BaseProps,
	BaseState,
	baseZIndex,
	borderStyle,
	boxStyle,
	Color,
	ColorScheme,
	defaultSize,
	Direction,
	disabled,
	DisabledCSS,
	fontStyle,
	FontStyle,
	getDefaultBaseProps,
	getDefaultBaseState,
	getDefaultWrapperProps,
	getTheme,
	getThemeList,
	invisible,
	InvisibleCSS,
	Justify,
	lineHeightStyle,
	Location,
	locationStyle,
	rectStyle,
	setTheme,
	Sizes,
	Sizing,
	SortOrder,
	Styles,
	Styling,
	Theme,
	ThemeProps,
	Wrapper,
	WrapperProps
};
