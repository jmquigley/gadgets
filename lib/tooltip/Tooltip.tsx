/**
 * A tooltip is a text popup element associated with a control.  It
 * is used to give help or feedback to the user on the use of that
 * control.  This `Tooltip` component is embedded within another
 * widget.  The *parent* widget contains an id.  The `Tooltip` is
 * given the same id as the *parent* prop.  When the *mouseenter* event
 * is invoked on the parent component the tooltip is displayed.  When
 * the *mouseleave* event occurs the tooltip is hidden.
 *
 * Note that the tooltip is set by position *absolute*.  The container
 * that will hold the `Tooltip` component must be position *relative*
 * or the  component will be placed as absolute from the beginning of the
 * document.
 *
 * #### Examples:
 *
 * ```javascript
 * import {Tooltip} from 'gadgets';
 *
 * <div id="uniqueId">
 *     ...
 *     <Tooltip location={Location.topRight} parent="uniqueId">tooltip string</Tooltip>
 * </div>
 * ```
 *
 * #### Events
 * None
 *
 * #### Styles
 * - `ui-tooltip` - Applied to the top level `<div>` in the control
 * - `ui-tooltip-content` - Applied to the container around the content
 * (text) of the tooltip.
 *
 * #### Properties
 * - `color: string ('white')` - the color of the tooltip text
 * - `backgroundColor: string ('gray')` - the color of the containing
 * box
 * - `location: Location (Location.middleRight)` - when the tooltip is
 * displayed this is the side of the parent control where it will be
 * displayed.
 * - `parent: {string} ('')` - The id of the component where this tooltip
 * will be applied.
 *
 * @module Tooltip
 */

// const debug = require('debug')('gadgets.Tooltip');

import autobind from "autobind-decorator";
import * as React from "react";
import {
	BaseComponent,
	BaseProps,
	BaseState,
	baseZIndex,
	Direction,
	fontStyle,
	getDefaultBaseProps,
	getDefaultBaseState,
	getTheme,
	Location,
	Wrapper
} from "../shared";
import styled, {css} from "../shared/themed-components";
import {Triangle} from "../triangle";

export interface TooltipProps extends BaseProps {
	parent?: any;
}

export function getDefaultTooltipProps(): TooltipProps {
	const theme = getTheme();

	return {
		...getDefaultBaseProps(),
		location: Location.top,
		obj: "Tooltip",
		parent: null,
		style: {
			color: theme.tooltipForegroundColor,
			backgroundColor: theme.tooltipBackgroundColor
		}
	};
}

export interface TooltipState extends BaseState {
	show?: boolean;
}

export function getDefaultTooltipState(): TooltipState {
	return {...getDefaultBaseState(), show: false};
}

const Bottom: any = css`
	top: 100%;
	left: 50%;
	transform: translateX(-50%);
`;

const BottomLeft: any = css`
	top: 100%;
	left: 25%;
	transform: translateX(-50%);
`;

const BottomRight: any = css`
	top: 100%;
	left: 75%;
	transform: translateX(-50%);
`;

const MiddleLeft: any = css`
	top: 50%;
	right: 100%;
	transform: translateY(-50%);
`;

const Middle: any = css`
	top: 50%;
	left: 100%;
	transform: translateY(-50%);
`;

const Top: any = css`
	bottom: 100%;
	left: 50%;
	transform: translateX(-50%);
`;

const TopLeft: any = css`
	bottom: 100%;
	left: 25%;
	transform: translateX(-50%);
`;

const TopRight: any = css`
	bottom: 100%;
	left: 75%;
	transform: translateX(-50%);
`;

const TriangleBottom: any = `
	position: absolute;
	top: -0.75em;
	left: 50%;
	transform: translateX(-50%);
`;

const TriangleBottomLeft: any = css`
	position: absolute;
	top: -0.75em;
	left: 75%;
	transform: translateX(-50%);
`;

const TriangleBottomRight: any = css`
	position: absolute;
	top: -0.75em;
	left: 25%;
	transform: translateX(-50%);
`;

const TriangleMiddleLeft: any = css`
	position: absolute;
	top: 50%;
	right: -0.75em;
	transform: translateY(-50%) rotate(90deg);
`;

const TriangleMiddleRight: any = css`
	position: absolute;
	top: 50%;
	left: -0.75em;
	transform: translateY(-50%) rotate(270deg);
`;

const TriangleTop: any = css`
	position: absolute;
	bottom: -0.75em;
	left: 50%;
	transform: translateX(-50%) rotate(180deg);
`;

const TriangleTopLeft: any = css`
	position: absolute;
	bottom: -0.75em;
	left: 75%;
	transform: translateX(-50%) rotate(180deg);
`;

const TriangleTopRight: any = css`
	position: absolute;
	bottom: -0.75em;
	left: 25%;
	transform: translateX(-50%) rotate(180deg);
`;

const Hide: any = css`
	opacity: 0;
	z-index: calc(${baseZIndex} - 1);
`;

const Show: any = css`
	opacity: 1;
	z-index: calc(${baseZIndex} + 1);
`;

const StyledTriangle: any = styled(Triangle)`
	${(props: TooltipProps) => {
		switch (props.location) {
			case Location.topLeft:
				return TriangleTopLeft;
			case Location.topRight:
				return TriangleTopRight;
			case Location.middleLeft:
				return TriangleMiddleLeft;
			case Location.middle:
			case Location.middleRight:
				return TriangleMiddleRight;
			case Location.bottomLeft:
				return TriangleBottomLeft;
			case Location.bottom:
				return TriangleBottom;
			case Location.bottomRight:
				return TriangleBottomRight;

			default:
			case Location.top:
				return TriangleTop;
		}
	}};
`;

const TootipContentView: any = styled.span`
	${(props: TooltipProps) => props.sizing && fontStyle[props.sizing]}
`;

const TooltipView: any = styled.div`
	border-radius: 10px;
	flex: none;
	line-height: initial;
	padding: 0.5em 1em;
	pointer-events: none;
	position: absolute;
	text-align: initial;
	transition: opacity 0.75s ease;
	width: ${(props: TooltipProps) => props.width};

	${(props: TooltipProps) => {
		switch (props.location) {
			case Location.topLeft:
				return TopLeft;
			case Location.topRight:
				return TopRight;
			case Location.middleLeft:
				return MiddleLeft;
			case Location.middle:
			case Location.middleRight:
				return Middle;
			case Location.bottomLeft:
				return BottomLeft;
			case Location.bottom:
				return Bottom;
			case Location.bottomRight:
				return BottomRight;

			default:
			case Location.top:
				return Top;
		}
	}};

	${(props: TooltipProps) => (props.visible ? Show : Hide)}
	${(props: TooltipProps) => props.sizing && fontStyle[props.sizing]}
`;

export class Tooltip extends BaseComponent<TooltipProps, TooltipState> {
	public static readonly defaultProps: TooltipProps = getDefaultTooltipProps();
	public state: TooltipState = getDefaultTooltipState();

	constructor(props: TooltipProps) {
		super(props, "ui-tooltip", Tooltip.defaultProps.style);
	}

	@autobind
	private handleMouseEnter() {
		this.setState({show: true});
	}

	@autobind
	private handleMouseLeave() {
		this.setState({show: false});
	}

	public componentDidMount() {
		const parent = document.getElementById(this.props.parent);
		if (parent) {
			parent.addEventListener("mouseenter", this.handleMouseEnter);
			parent.addEventListener("mouseleave", this.handleMouseLeave);
		}
	}

	public componentWillUnmount() {
		const parent = document.getElementById(this.props.parent);
		if (parent) {
			parent.removeEventListener("mouseenter", this.handleMouseEnter);
			parent.removeEventListener("mouseleave", this.handleMouseLeave);
		}
	}

	public render() {
		this.updateClassName();

		let direction: Direction = Direction.left;

		switch (this.props.location) {
			case Location.topLeft:
			case Location.top:
			case Location.topRight:
				direction = Direction.down;
				break;

			case Location.middleLeft:
				direction = Direction.right;
				break;

			case Location.bottomLeft:
			case Location.bottom:
			case Location.bottomRight:
				direction = Direction.up;
				break;
		}

		let str: string = "";
		if (this.props.children) {
			str = React.Children.map(this.props.children, (child: any) => {
				return String(child);
			}).join(" ");
		}

		return (
			<Wrapper {...this.props}>
				<TooltipView
					className={this.className}
					location={this.props.location}
					sizing={BaseComponent.prev(this.props.sizing).type}
					style={this.state.style}
					visible={this.state.show}
					width={str.length < 25 ? "unset" : "10em"}
				>
					<TootipContentView
						className='ui-tooltip-content'
						sizing={BaseComponent.prev(this.props.sizing).type}
						style={this.state.style}
					>
						{str}
					</TootipContentView>
					<StyledTriangle
						location={this.props.location}
						direction={direction}
						sizing={BaseComponent.prev(this.props.sizing).type}
						style={{
							fill: this.state.style["backgroundColor"],
							stroke: this.state.style["backgroundColor"]
						}}
					/>
				</TooltipView>
			</Wrapper>
		);
	}
}

/**
 * Creates a tooltip object for use within a control.  It will check the given
 * props for a tooltip string.  If it has one, it will create the object and
 * return it.  If it doesn't have it, then NULL is returned.
 * @param props {any} and object representing the props used to generate the
 * tooltip.
 * @return {Tooltip} a new Tooltip reference if there is a given tooltip string
 * otherwise null is returned.
 */
export function tooltip(id: string, props: any) {
	if (props["tooltip"] && !props["notooltip"]) {
		return (
			<Tooltip parent={id} sizing={props["sizing"]}>
				{props["tooltip"]}
			</Tooltip>
		);
	}

	return null;
}

export default Tooltip;
