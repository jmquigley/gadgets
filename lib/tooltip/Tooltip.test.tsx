import {EnumValues as ev} from "enum-values";
import {mount, shallow} from "enzyme";
import assert from "power-assert";
import * as React from "react";
import {Location} from "../shared";
import Tooltip from "./Tooltip";

for (const location of ev.getNames(Location)) {
	test(`Creation of Tooltip in position ${location}`, () => {
		const ctl = mount(<Tooltip location={location}>Test Text</Tooltip>);

		assert(ctl);
		expect(ctl).toMatchSnapshot();
	});
}

test("Test creation of Tooltip location none", () => {
	const ctl = mount(<Tooltip location={Location.none}>Test Text</Tooltip>);
	expect(ctl).toBeDefined();
	expect(ctl).toMatchSnapshot();
});

test("Test creation of a Tooltip control", () => {
	const ctl = shallow(
		<Tooltip className='test-class' location={Location.middleRight}>
			<span>Test Text</span>
		</Tooltip>
	);

	expect(ctl).toBeDefined();
	expect(ctl).toMatchSnapshot();
});

test("Test disabling of a Tooltip", () => {
	const ctl = shallow(<Tooltip disabled={true} />);

	expect(ctl).toBeDefined();
	expect(ctl).toMatchSnapshot();
});

test("Test making a Tooltip invisible", () => {
	const ctl = shallow(<Tooltip visible={false} />);

	expect(ctl).toBeDefined();
	expect(ctl).toMatchSnapshot();
});
