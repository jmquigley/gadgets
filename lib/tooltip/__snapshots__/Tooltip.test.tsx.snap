// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creation of Tooltip in position bottom 1`] = `
<Tooltip
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  left={null}
  location="bottom"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="Tooltip"
  padding={null}
  parent={null}
  ripple={false}
  selected={false}
  sizing="normal"
  style={
    Object {
      "backgroundColor": "#708090",
      "color": "#ffffff",
    }
  }
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Creation of Tooltip in position bottomLeft 1`] = `
<Tooltip
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  left={null}
  location="bottomLeft"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="Tooltip"
  padding={null}
  parent={null}
  ripple={false}
  selected={false}
  sizing="normal"
  style={
    Object {
      "backgroundColor": "#708090",
      "color": "#ffffff",
    }
  }
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Creation of Tooltip in position bottomRight 1`] = `
<Tooltip
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  left={null}
  location="bottomRight"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="Tooltip"
  padding={null}
  parent={null}
  ripple={false}
  selected={false}
  sizing="normal"
  style={
    Object {
      "backgroundColor": "#708090",
      "color": "#ffffff",
    }
  }
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Creation of Tooltip in position left 1`] = `
<Tooltip
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  left={null}
  location="left"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="Tooltip"
  padding={null}
  parent={null}
  ripple={false}
  selected={false}
  sizing="normal"
  style={
    Object {
      "backgroundColor": "#708090",
      "color": "#ffffff",
    }
  }
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Creation of Tooltip in position middle 1`] = `
<Tooltip
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  left={null}
  location="middle"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="Tooltip"
  padding={null}
  parent={null}
  ripple={false}
  selected={false}
  sizing="normal"
  style={
    Object {
      "backgroundColor": "#708090",
      "color": "#ffffff",
    }
  }
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Creation of Tooltip in position middleLeft 1`] = `
<Tooltip
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  left={null}
  location="middleLeft"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="Tooltip"
  padding={null}
  parent={null}
  ripple={false}
  selected={false}
  sizing="normal"
  style={
    Object {
      "backgroundColor": "#708090",
      "color": "#ffffff",
    }
  }
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Creation of Tooltip in position middleRight 1`] = `
<Tooltip
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  left={null}
  location="middleRight"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="Tooltip"
  padding={null}
  parent={null}
  ripple={false}
  selected={false}
  sizing="normal"
  style={
    Object {
      "backgroundColor": "#708090",
      "color": "#ffffff",
    }
  }
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Creation of Tooltip in position none 1`] = `
<Tooltip
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="Tooltip"
  padding={null}
  parent={null}
  ripple={false}
  selected={false}
  sizing="normal"
  style={
    Object {
      "backgroundColor": "#708090",
      "color": "#ffffff",
    }
  }
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Creation of Tooltip in position right 1`] = `
<Tooltip
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  left={null}
  location="right"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="Tooltip"
  padding={null}
  parent={null}
  ripple={false}
  selected={false}
  sizing="normal"
  style={
    Object {
      "backgroundColor": "#708090",
      "color": "#ffffff",
    }
  }
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Creation of Tooltip in position top 1`] = `
<Tooltip
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  left={null}
  location="top"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="Tooltip"
  padding={null}
  parent={null}
  ripple={false}
  selected={false}
  sizing="normal"
  style={
    Object {
      "backgroundColor": "#708090",
      "color": "#ffffff",
    }
  }
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Creation of Tooltip in position topLeft 1`] = `
<Tooltip
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  left={null}
  location="topLeft"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="Tooltip"
  padding={null}
  parent={null}
  ripple={false}
  selected={false}
  sizing="normal"
  style={
    Object {
      "backgroundColor": "#708090",
      "color": "#ffffff",
    }
  }
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Creation of Tooltip in position topRight 1`] = `
<Tooltip
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  left={null}
  location="topRight"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="Tooltip"
  padding={null}
  parent={null}
  ripple={false}
  selected={false}
  sizing="normal"
  style={
    Object {
      "backgroundColor": "#708090",
      "color": "#ffffff",
    }
  }
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Test creation of Tooltip location none 1`] = `
<Tooltip
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="Tooltip"
  padding={null}
  parent={null}
  ripple={false}
  selected={false}
  sizing="normal"
  style={
    Object {
      "backgroundColor": "#708090",
      "color": "#ffffff",
    }
  }
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Test creation of a Tooltip control 1`] = `
<Wrapper
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className="test-class"
  color={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  left={null}
  location="middleRight"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="Tooltip"
  onError={[Function]}
  padding={null}
  parent={null}
  reset={false}
  ripple={false}
  selected={false}
  sizing="normal"
  style={
    Object {
      "backgroundColor": "#708090",
      "color": "#ffffff",
    }
  }
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
>
  <Tooltip__TooltipView
    className="ui-tooltip test-class"
    location="middleRight"
    sizing="small"
    style={
      Object {
        "backgroundColor": "#708090",
        "color": "#ffffff",
      }
    }
    visible={false}
    width="unset"
  >
    <Tooltip__TootipContentView
      className="ui-tooltip-content"
      sizing="small"
      style={
        Object {
          "backgroundColor": "#708090",
          "color": "#ffffff",
        }
      }
    >
      [object Object]
    </Tooltip__TootipContentView>
    <Tooltip__StyledTriangle
      direction="left"
      location="middleRight"
      sizing="small"
      style={
        Object {
          "fill": "#708090",
          "stroke": "#708090",
        }
      }
    />
  </Tooltip__TooltipView>
</Wrapper>
`;

exports[`Test disabling of a Tooltip 1`] = `
<Wrapper
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={true}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  left={null}
  location="top"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="Tooltip"
  onError={[Function]}
  padding={null}
  parent={null}
  reset={false}
  ripple={false}
  selected={false}
  sizing="normal"
  style={
    Object {
      "backgroundColor": "#708090",
      "color": "#ffffff",
    }
  }
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
>
  <Tooltip__TooltipView
    className="ui-tooltip nohover"
    location="top"
    sizing="small"
    style={
      Object {
        "backgroundColor": "#708090",
        "color": "#ffffff",
      }
    }
    visible={false}
    width="unset"
  >
    <Tooltip__TootipContentView
      className="ui-tooltip-content"
      sizing="small"
      style={
        Object {
          "backgroundColor": "#708090",
          "color": "#ffffff",
        }
      }
    />
    <Tooltip__StyledTriangle
      direction="down"
      location="top"
      sizing="small"
      style={
        Object {
          "fill": "#708090",
          "stroke": "#708090",
        }
      }
    />
  </Tooltip__TooltipView>
</Wrapper>
`;

exports[`Test making a Tooltip invisible 1`] = `
<Wrapper
  backgroundColor={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  contentEditable={false}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  id={null}
  left={null}
  location="top"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noedit={false}
  nohover={false}
  notheme={false}
  notooltip={false}
  obj="Tooltip"
  onError={[Function]}
  padding={null}
  parent={null}
  reset={false}
  ripple={false}
  selected={false}
  sizing="normal"
  style={
    Object {
      "backgroundColor": "#708090",
      "color": "#ffffff",
    }
  }
  testing={true}
  theme={null}
  tooltip={null}
  top={null}
  visible={false}
  width={null}
  xcss={null}
>
  <Tooltip__TooltipView
    className="ui-tooltip"
    location="top"
    sizing="small"
    style={
      Object {
        "backgroundColor": "#708090",
        "color": "#ffffff",
      }
    }
    visible={false}
    width="unset"
  >
    <Tooltip__TootipContentView
      className="ui-tooltip-content"
      sizing="small"
      style={
        Object {
          "backgroundColor": "#708090",
          "color": "#ffffff",
        }
      }
    />
    <Tooltip__StyledTriangle
      direction="down"
      location="top"
      sizing="small"
      style={
        Object {
          "fill": "#708090",
          "stroke": "#708090",
        }
      }
    />
  </Tooltip__TooltipView>
</Wrapper>
`;

exports[`Test retrieval of Tooltip props object 1`] = `
Object {
  "backgroundColor": null,
  "borderColor": null,
  "bottom": null,
  "children": null,
  "className": null,
  "color": null,
  "contentEditable": false,
  "controlled": true,
  "direction": "right",
  "disabled": false,
  "err": null,
  "errorMessage": null,
  "focus": false,
  "height": null,
  "id": null,
  "left": null,
  "location": "top",
  "maxHeight": null,
  "maxWidth": null,
  "minHeight": null,
  "minWidth": null,
  "noedit": false,
  "nohover": false,
  "notheme": false,
  "notooltip": false,
  "obj": "Tooltip",
  "padding": null,
  "parent": null,
  "ripple": false,
  "selected": false,
  "sizing": "normal",
  "style": Object {
    "backgroundColor": "#708090",
    "color": "#ffffff",
  },
  "testing": true,
  "theme": null,
  "tooltip": null,
  "top": null,
  "visible": true,
  "width": null,
  "xcss": null,
}
`;

exports[`Test retrieval of Tooltip props object 2`] = `
Object {
  "children": null,
  "show": false,
  "sizing": "normal",
  "style": Object {},
}
`;
